code/todo_list.rb

```diff
@@ -0,0 +1,72 @@
+class TodoItem
+	attr_accessor :description
+
+	def initialize(description, done=false)
+		@description = description
+		@done = done
+	end
+
+	def done?
+		@done
+	end
+
+	def done!
+		@done = true
+	end
+
+end
+
+class TodoList
+	attr_reader :name, :color, :items
+
+	def initialize(name, color={})
+		@name = name
+		@color = color[:color]
+		@items = []
+	end
+
+	def add(item)
+		if item.class == TodoItem
+			@items.push(item)
+		else
+			@items.push(TodoItem.new(item))
+		end
+	end
+
+	def items_pending
+		@items_pending = []	
+		@items.each do |x|
+			if !x.done?
+				@items_pending.push(x)
+			end
+		end
+		@items_pending
+	end
+
+	def items_done
+		@items_done = []
+		@items.each do |x|
+			if x.done?
+				@items_done.push(x)
+			end
+		end
+		@items_done
+	end
+
+	def find_by_description(description)
+		@items.each do |x|
+			if x.description == description
+				x
```

Comment:
Gah, aici a fost testul meu oribil.

Ce se intampla aici e ca find_by_description iti returneaza valoarea returnata de ultima instructiune executata, respectiv .each.

.each, in cazul lui array, returneaza array-ul peste care a iterat, adica in acest caz, ```@items```.

Implementarea asta ar merge in cazul in care aici ai pune un return explicit, care in cazul ala ar si iesi din toata metoda, eventual urmat de un nil dupa .each, astfel incat valoarea returnata daca nu gaseste ceva sa fie nil, nu array-ul in sine.

De asemenea, dacă îl faci pe ```find_by_description``` așa, poți să-ți simplifici și ```set_as_done```